lib/index.js

```diff
@@ -46,57 +47,81 @@ module.exports = function attachCredentialsPluginFactory(pluginOptions) {
 
   return {
     name: PLUGIN_NAME,
-    load: (req, origParams, callback) => {
-      // create a copy of provided request parameters (remove user prop if requested in `options`)
-      const reqParams = extractRequestparams(origParams);
-      const { [userPropName]: user } = origParams;
-
-      if (!user) {
-        debug('No "%s" prop found in request params, skipping plugin operations', userPropName, origParams);
-        // [bk] @TODO: should we use `reqParams` here instead?
-        // That would mean we potentially pass back params without
-        // the user prop and without credentials
-        callback(null, origParams);
-        return;
-      }
+    onRequest: [{
+      phaseName: REQUEST_PHASE_NAME,
+      handler: (ctx, next) => {
+        const { options } = ctx;
+        const {
```

Comment:
seems like code that is better suited in `oniyi-http-client`... and there, actual plugin execution should be skipped.
If the plugin wants to provide `skip` switch via `options`, it should use a different name (e.g. `skipCredentialsPlugin`). PhaseName must not be used. For now, I'd completely remove this feature. Feel free to provide a separate PR to add skipping logic